graph/telemetry/istio/appender/gateway_hostnames.go

```diff
@@ -0,0 +1,133 @@
+
+	log.Errorf("Found ingress GW: %d, %v", len(globalInfo.IngressWorkloads), globalInfo.IngressWorkloads)
+	istioAppLabelName := config.Get().IstioLabels.AppLabelName
+
+	gatewayInNamespace := false
+	if ingressWorkloadsList, ok := globalInfo.IngressWorkloads[namespaceInfo.Namespace]; ok {
+		for _, gw := range ingressWorkloadsList {
+			for _, node := range trafficMap {
+				if node.App == gw.Labels[istioAppLabelName] && node.Namespace == namespaceInfo.Namespace {
+					node.Metadata[graph.IsIngressGw] = true
+					ingressNodeMapping[&gw] = append(ingressNodeMapping[&gw], node)
+					gatewayInNamespace = true
+				}
+			}
+		}
+	}
+
+	log.Errorf("gatewayInNamespace for ns %s = %v", namespaceInfo.Namespace, gatewayInNamespace)
+
+	// If there is any ingress gateway node in the processing namespace, find Gateway CRDs and
+	// match them against gateways in the graph.
+	if gatewayInNamespace {
+		gatewaysCrds := getIstioGatewayResources(globalInfo, namespaceInfo)
+
+		for _, gwCrd := range gatewaysCrds {
+			gwSelector := labels.Set(gwCrd.Spec.Selector).AsSelector()
+			for gw, nodes := range ingressNodeMapping {
+				if gwSelector.Matches(labels.Set(gw.Labels)) {
+
+					// If we are here, the GatewayCrd selects the Gateway workload.
+					// So, all node graphs associated with the GW workload should be listening
+					// requests for the hostnames listed in the GatewayCRD.
+
+					// Let's extract the hostnames and add them to the node metadata.
+					for _, node := range nodes {
+						gwServers := gwCrd.Spec.Servers.([]interface{})
+						var hostnames []string
+
+						for _, gwServer := range gwServers {
+							gwServerMap := gwServer.(map[string]interface{})
+							gwHosts := gwServerMap["hosts"].([]interface{})
+							for _, gwHost := range gwHosts {
+								hostnames = append(hostnames, gwHost.(string))
+							}
+						}
+
+						// Metadata format: { gatewayName => array of hostnames }
+						node.Metadata[graph.Gateways] = graph.GatewaysMetadata{
+							gwCrd.Metadata.Name: hostnames,
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+func (GatewayHostnamesAppender) Name() string {
+	return GatewayHostnamesAppenderName
+}
+
+func getIngressGatewaysWorkloads(globalInfo *graph.AppenderGlobalInfo) {
+	nsList, nsErr := globalInfo.Business.Namespace.GetNamespaces()
+	graph.CheckError(nsErr)
+
+	ingressWorkloads := make(map[string][]models.WorkloadListItem)
+	for _, namespace := range nsList {
+		wList, err := globalInfo.Business.Workload.GetWorkloadList(namespace.Name)
+		graph.CheckError(err)
+
+		log.Errorf("getIngressGatewaysWorkloads in ns [%s] wList: %d // %v", namespace.Name, len(wList.Workloads), wList)
+
+		// Find Ingress Gateway deployments
+		for _, workload := range wList.Workloads {
+			log.Errorf("Workload %s type is %s. Labels are: ", workload.Name, workload.Type)
+			if workload.Type == "Deployment" {
+				if labelValue, ok := workload.Labels["operator.istio.io/component"]; ok && labelValue == "IngressGateways" {
```

Comment:
Hi @israel-hdez, a coupe of early comments in case it helps you with the work:
- In advanced scenarios, users tend to create their own gateways like [1], it seems that label may not be present if they install without operator.
- I think the case for Gateways Egress would be similar [2], so probably you can cover it in the same work.
- Also, there are multiple filters for kubernetes/Istio objects, located under kubernetes/filters.go, or in this case, you could  address a filter on the WorkloadList located in the model, then that would be reused better. 

[1] https://istio.io/latest/docs/setup/additional-setup/gateway/
[2] https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/

Reply:
I'll fix the last 2 points in a new commit.

About the first point, I'm not sure how we could cover custom gateways without adding new config.
There are no mandatory labels for gateways. Without adding more config to the KialiCR, I don't think it is possible to reliably determine which workloads are gateways.

I tend to prefer usage of labels set in the Istio operator/helm, because I see them as the labels by convention. If people are using a highly customized installation, I'd prefer to see a community contribution to cover that case.